python/tchannel/scheme.py

```diff
@@ -41,10 +62,18 @@ def __init__(self):
     def type(self):
         return self._type
 
-    def deserialize(self, obj):
+    def deserialize_header(self, obj):
+        assert obj
```

Comment:
I think it's okay to use `assert`s for what are essentially 'programmer errors'. Things like invalid state, for example.